### PR TITLE
Communities

### PR DESCRIPTION
**Pull request by vurpo**
I made a new pull request for this one because I had branch confusion, and now I merged in the new changes on the original branch instead of communities-merged. (It doesn't seem like you can change the source branch for a PR on GitHub.)

### CODE REVIEW

**Comment by mujx** [comment]
Could you take a look at the ci failures? Also the `.vscode` dir should be excluded.

File: src/MatrixClient.cc [hunk 1]
```diff
@@ -315,20 +314,20 @@ MatrixClient::sendRoomMessage(mtx::events::MessageType ty,
 
                 if (!json.isObject()) {
                         qDebug() << "Send message response is not a JSON object";
-                        emit messageSendFailed(roomid, txnId);
```
Comment (mujx) [inline src/MatrixClient.cc]: We need to keep this.

File: src/MatrixClient.cc [hunk 2]
```diff
@@ -315,20 +314,20 @@ MatrixClient::sendRoomMessage(mtx::events::MessageType ty,
...
                         return;
                 }
 
                 auto object = json.object();
 
                 if (!object.contains("event_id")) {
                         qDebug() << "SendTextMessage: missing event_id from response";
-                        emit messageSendFailed(roomid, txnId);
```
Comment (mujx) [inline src/MatrixClient.cc]: We also need this.

File: src/MatrixClient.cc [hunk 3]
```diff
@@ -315,20 +314,20 @@ MatrixClient::sendRoomMessage(mtx::events::MessageType ty,
...
                         return;
                 }
 
                 emit messageSent(object.value("event_id").toString(), roomid, txnId);
         });
+
+        incrementTransactionId();
```
Comment (mujx) [inline src/MatrixClient.cc]: This can be removed.

File: src/MainWindow.cc [hunk 4]
```diff
@@ -128,6 +128,15 @@ MainWindow::MainWindow(QWidget *parent)
         }
 }
 
+void
```
Comment (mujx) [inline src/MainWindow.cc]: This is old code and can be removed.

File: include/MainWindow.h [hunk 5]
```diff
@@ -47,6 +47,7 @@ class MainWindow : public QMainWindow
 
 protected:
         void closeEvent(QCloseEvent *event);
+        void keyPressEvent(QKeyEvent *event);
```
Comment (mujx) [inline include/MainWindow.h]: This is old code and can be removed.

File: src/MatrixClient.cc (hunk 3 above)
Comment (vurpo) [reply src/MatrixClient.cc]: Yeah, git was very unhelpful when it showed the merge conflict for this file

**Pull request merged**